### ISSUE TITLE
"An error occured when attempting to save"

### ISSUE THREAD

**Issue by Vevasap** [talmolab/sleap#2622]
I transferred a project from one computer to another and started to correct predictions and re-train. After my frames were labeled, I went to train and was told to save the project first. Okay, but when I go to save I am given this:

"An error occured when attempting to save:
Object of type int64 is not JSON serializable

Try saving your project with a different file name or in a different format."

This also happens if I try to export labels package:

"Failed to export labels package:
Object of type int64 is not JSON serializable"

Would be great to be able to save or train without having to close the software and lose my afternoon of labeling...

<img width="413" height="160" alt="Image" src="https://github.com/user-attachments/assets/8d8b1d06-dfd0-410d-871a-3ac24ca9d318" />

## System diagnostics

<details><summary>Output of <code>sleap doctor</code></summary>
SLEAP System Diagnostics
========================
Generated: 2026-02-20 18:05:02

[Platform]
  OS:        Windows 10
  Platform:  Windows-10-10.0.19045-SP0
  Machine:   AMD64
  Processor: Intel64 Family 6 Model 85 Stepping 4, GenuineIntel
  RAM:       14.0 GB / 63.7 GB
  Disk:      418.3 GB / 465.2 GB

[Python]
  Version:     3.13.11
  Executable:  C:\Users\Akino\anaconda3\envs\sleap\python.exe
  Prefix:      C:\Users\Akino\anaconda3\envs\sleap

[Conda]
  Status:             ACTIVE
  Environment:        sleap
  Prefix:             C:\Users\Akino\anaconda3\envs\sleap

[GPU / CUDA]
  NVIDIA Driver: 572.16 (CUDA 12.8)
  GPU 0:         NVIDIA TITAN Xp (11186 MB free / 12288 MB)
  PyTorch:       v2.10.0+cu128 (CUDA 12.8)

[Packages]
  sleap:          v1.6.1 (pip)
  sleap-io:       v0.6.4 (pip)
  sleap-nn:       v0.1.0 (pip)
  numpy:          v2.4.2 (pip)
  h5py:           v3.15.1 (pip)
  PySide6:        v6.10.2 (pip)
  opencv-python:  v4.13.0.92 (pip)
  torch:          v2.10.0+cu128 (pip)
  imageio:        v2.37.2 (pip)
  imageio-ffmpeg: v0.6.0 (pip)
  av:             v15.1.0 (pip)

**Comment by talmo** [comment]
Hi @Vevasap,

Oh no, sorry about the data loss!! We try hard to make sure that doesn't happen, but sometimes these bugs slip through the cracks. I always recommend saving often so you can catch these early and be able to revert changes if something goes wrong (even when there aren't bugs).

Do you mind sharing the SLP file you were working with via https://slp.sh? We'll take a look on our end and investigate.

Thanks and sorry about your afternoon of lost labeling!!

Talmo

**Comment by alicup29** [comment]
Hey @Vevasap ! 

The likely source of this issue is that `SuggestionFrame.frame_idx` wasn't coercing `numpy int64` to `Python int`, which caused the JSON serialization error you saw when saving. We addressed this issue afterward in [sleap-io PR #350](https://github.com/talmolab/sleap-io/pull/350). Upgrading to the latest version of SLEAP should resolve this. 

I'll be closing this issue for now, but if you run into anymore related issues, please feel free to open a new issue! We apologize again for the lost labeling session!

Thanks,
Amick
